nat-lab/tests/test_upnp_connection.py

```diff
@@ -14,9 +14,31 @@
 )
 from utils.connection_util import ConnectionTag
 from utils.ping import ping
+from utils.process import ProcessExecError
 from utils.router import new_router, IPStack
 
 
+async def execute_upnpc_with_retry(connection, timeout=10.0):
+    start_time = asyncio.get_event_loop().time()
+    while True:
+        try:
+            cmd = await connection.create_process(["upnpc", "-i", "-l"]).execute()
```

Comment:
I think we can simplify it abit. No need to check return code, since it will be zero or 137 (sigkill which is mostly caused by leaving stack == test cleanup), otherwise exception will be raised.
```
async def execute_upnpc_with_retry(connection, timeout=10.0):
    start_time = asyncio.get_event_loop().time()
    while True:
        try:
            return await connection.create_process(["upnpc", "-i", "-l"]).execute()
        except ProcessExecError as e:
            if asyncio.get_event_loop().time() - start_time >= timeout:
                print(f"Timeout while waiting for upnpc to start: {str(e)}")
                raise

        await asyncio.sleep(1.0)
```

Reply:
Fixed